### DISCUSSION TITLE
Those annoying “you need a jdk” dialogs

### DISCUSSION THREAD

**Discussion by dabrahams** [Homebrew/discussions#460] [Everyday usage]
Often when I'm installing a package I'll get a dialog about installing java.  I've learned that I can ignore these dialogs and everything *seems* to work fine.  What functionality am I missing out on by ignoring the dialog?  Is there any way to suppress them?

**Comment by SMillerDev** [comment]
I've never seen such dialogs. Do you have a screenshot?

**Reply by dabrahams** [reply]
No but I'll post one next time it comes up.

**Reply by bgs113** [reply]
I'm also seeing this any time I run `brew doctor`. Here's the dialog.

<img width="488" alt="image" src="https://user-images.githubusercontent.com/10234957/107092823-d7b54280-67c9-11eb-82f5-b2eb9e44c37c.png">

See also https://github.com/Homebrew/brew/issues/10491

**Reply by panolens** [reply]
> I'm also seeing this any time I run `brew doctor`. Here's the dialog.
> 
> <img alt="image" width="488" src="https://user-images.githubusercontent.com/10234957/107092823-d7b54280-67c9-11eb-82f5-b2eb9e44c37c.png">
> 
> See also [Homebrew/brew#10491](https://github.com/Homebrew/brew/issues/10491)

..seems to have been fiexed in 3.0.1? -> I don't see it anymore.

**Comment by gromgit** [comment]
I work on both macOS and Linux, and ever so often, I forget which terminal I'm working on and do an `apt install XYZ` on macOS. `apt` is the system packager on Debian/Ubuntu, but the Java "annotation processing tool" on macOS, and since I've never "bound" a JDK to macOS, that inevitably pops this dialog box up:

![](https://aws1.discourse-cdn.com/standard17/uploads/brew/original/2X/9/9fd0ead62d7771e6bd0f1630e72a9148c424a7c9.png)

The command also prints `No Java runtime present, requesting install.` on the offending terminal.

Outside of that, I've never managed to trigger a "you need a JDK" dialog box while installing software on macOS, so if you're in the same boat, the simplest solution is to stop doing it. :grin:

**Comment by cryo-kronos** [comment]
So, this dialog has suddenly appeared on my macOS Mojave system after upgrading to Homebrew 3.0 via "brew upgrade". It only appears when I run "brew doctor". And it never popped up before the upgrade. 

In the JDK alert dialog window, there is a "More Info" button. Clicking it takes me to the Adobe Flash EOL page (https://www.adobe.com/products/flashplayer/end-of-life.html). I don't have Flash installed on my system. And I'm not sure what JDK has to do with Flash.

Any suggestions on how to fix this or do we need to wait for a bugfix and hombrew update?

**Reply by Bo98** [reply]
The bug is being tracked here: Homebrew/brew#10491. You can subscribe to that for updates.

**Reply by panolens** [reply]
> So, this dialog has suddenly appeared on my macOS Mojave system after upgrading to Homebrew 3.0 via "brew upgrade". It only appears when I run "brew doctor". And it never popped up before the upgrade.
> 
> In the JDK alert dialog window, there is a "More Info" button. Clicking it takes me to the Adobe Flash EOL page (https://www.adobe.com/products/flashplayer/end-of-life.html). I don't have Flash installed on my system. And I'm not sure what JDK has to do with Flash.
> 
> Any suggestions on how to fix this or do we need to wait for a bugfix and hombrew update?

Same here!
